src/routes/upload/+page.svelte

```diff
@@ -74,18 +74,37 @@
         show_dropzone = false;
         dropzone_active = false;
 
-        const files = event.dataTransfer?.files as FileList;
+        const files = event.dataTransfer?.items as unknown as DataTransferItemList;
         const file_list_names = data_list.map((data) => {
             return data.file.name;
         });
 
-        Array.from(files).forEach((file) => {
-            if (Object.keys(file_whitelist).includes(file.type)) {
-                if (!file_list_names.includes(file.name)) {
-                    data_list = data_list.concat({ file: file });
-                }
+        Array.from(files).forEach(async (item) => {
+            const file = item.webkitGetAsEntry();
+            
+            if(file?.isDirectory){
+                const files = await scan_directory(file as FileSystemDirectoryEntry);
+                console.log("I cant see", files);
```

Comment:
`files` shows as `undefined`
but it consoles in `scan_directory` function

Reply:
Oh well, welcome to callback hell